crates/core/src/config.rs

```diff
@@ -1,18 +1,37 @@
 use std::path::Path;
 use std::{fmt, io};
 
+use toml;
+use toml_edit::{DocumentMut, Formatted, Item, Table, Value};
+
 use spacetimedb_lib::Address;
 use spacetimedb_paths::cli::{ConfigDir, PrivKeyPath, PubKeyPath};
 use spacetimedb_paths::server::{ConfigToml, MetadataTomlPath};
 
-#[derive(serde::Serialize, serde::Deserialize)]
-pub struct MetadataFile {
-    pub version: semver::Version,
-    pub edition: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub client_address: Option<Address>,
```

Comment:
nit: this diff is a little harder to review because several things moved without changing, like `MetadataFile`, `current_version`, and `impl ConfigFile`.

It's fine, I was able to review it, but in the future maybe those changes could be done separately?